certificates/src/lib.rs

```diff
@@ -108,6 +134,35 @@ impl CertificateValidator {
         // to use with verifying the private key.
         let cert = certs.swap_remove(0);
 
+        if let Some(hostname) = hostname {
+            let c_hostname = CString::new(hostname).map_err(|_| {
+                CertificateError::InvalidValidationHostname(
+                    hostname.to_string(),
+                )
+            })?;
+            if !cert
+                .valid_for_hostname(&c_hostname)
+                .map_err(CertificateError::ErrorValidatingHostname)?
+            {
+                let cert_description = cert
+                    .hostname_description()
+                    .map_err(CertificateError::ErrorValidatingHostname)?;
```

Comment:
Yeah, good call. I strongly suspect this error is impossible (i.e., if we can't get the hostnames out of the cert, I'm pretty sure `X509_check_host` would have already failed for the same reason), but in case it's not, this is fixed in 3ab3f03d